test/test.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from 'fs';
+import * as test from 'tape';
+import { ReferenceTranscoder } from './reference';
+
+const nModes = 19;
+const size: number = Number(process.env.GEN_SIZE) || 256;
+const nBlocks = 8 * size * size;
```

Comment:
Ditto.

Reply:
I'm not sure what you mean by 'more canonical', could you clarify? Note that the RGBA32 decoder example code deals with width/height explicitly when computing uncompressed length.

Reply:
`8` is an arbitrary number here. If we need to derive `nBlocks` from the texture's size, it should be like (to keep the value the same):
```
const width = 8192;
const height = 16384;
const nBlocks = ((width + 3) >> 2) * ((height + 3) >> 2);
```
In this case, it also would be better to use `width` and `height` in `ImageInfo` for the reference transcoder instead of hard-coded `4` and `4 * nBlocks`,

Reply:
Ok, yeah that's clearer, and it ironed out the test failure I'd been getting with RGBA32. 👍